Updated documentation for Site Sync for Ayon (#115)
ynput/ayon-documentation: website/docs/addon_site_sync_admin.md

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

Site Sync allows users and studios to synchronize published assets between multiple 'sites'. Site denotes a storage location,
which could be a physical disk, server, cloud storage. To be able to use site sync, it first needs to be configured. 

The general idea is that each user acts as an individual site and can download and upload any published project files when they are needed. that way, artist can have access to the whole project, but only every store files that are relevant to them on their home workstation. 

:::note

> At the moment site sync is only able to deal with publishes files (eg. in `publish` folder). No workfiles (in `work` folder) will be synchronized. We are working on making workfile synchronization possible as well. 

Reviewer: ```suggestion
At the moment site sync is only able to deal with published files (eg. in `publish` folder). **No workfiles (in `work` folder) will be synchronized.** We are working on making workfile synchronization possible as well.
```

:::

## Studio Settings

> To use synchronization, *Site Sync* needs to be enabled globally in **ayon+settings://sitesync** and available in Bundle.

Reviewer: ```suggestion
First of all, you need to include Site Sync Addon to your bundle, and to use synchronization, *Site Sync* needs to be enabled globally in **ayon+settings://sitesync**.
```

#### How to set SFTP site

- Add side with SFTP provider in Studio Settings

![Enable syncing and create site](assets/site_sync_sftp_system.png)

- In Projects setting enable Site Sync
- Configure SFTP connection and destination folder on a SFTP server (in screenshot `/upload`)
  
- if you want to force syncing between local and sftp site for all users, use combination `active site: local`, `remote site: NAME_OF_SFTP_SITE`
- if you want to allow only specific users to use SFTP syncing (external users, not located in the office), use `active site: studio`, `remote site: studio`. 

![Select active and remote site on a project](assets/site_sync_sftp_project_setting_not_forced.png)

- Each artist can decide and configure syncing from his/her local to SFTP via `Site Settings`

![Select active and remote site on a project](assets/site_sync_sftp_settings_local.png)
  
### Custom providers

> If a studio needs to use other services for cloud storage, or want to implement totally different storage providers, they can do so by writing their own provider plugin. We're working on a developer documentation, however, for now we recommend looking at `abstract_provider.py`and `gdrive.py` inside `ayon-sitesync/client/ayon-sitesync/providers` and using it as a template.

Reviewer: ```suggestion
If a studio needs to use other services for cloud storage, or want to implement totally different storage providers, they can do so by writing their own provider plugin. We're working on a developer documentation, however, for now we recommend looking at [`abstract_provider.py`](https://github.com/ynput/ayon-sitesync/blob/develop/client/ayon_sitesync/providers/abstract_provider.py) and [`gdrive.py`](https://github.com/ynput/ayon-sitesync/blob/develop/client/ayon_sitesync/providers/gdrive.py) inside `ayon-sitesync/client/ayon-sitesync/providers` and using it as a template.
```
